lib/hsa/mcwamp_hsa.cpp

```diff
@@ -5376,10 +5375,10 @@ HSACopy::syncCopyExt(hc::hcCommandKind copyDir, const hc::AmPointerInfo &srcPtrI
 
         // Get a signal and initialize it:
         std::pair<hsa_signal_t, int> ret = Kalmar::ctx.getSignal();
-        signal = ret.first;
-        signalIndex = ret.second;
+        _signal = ret.first;
+        _signalIndex = ret.second;
 
-        hsa_signal_store_relaxed(signal, 1);
+        hsa_signal_store_relaxed(_signal, 1);
```

Comment:
Should either consider using screlease or a release fence here, to ensure no reordering / fully observable precursors as far as the CP is concerned?

Reply:
These are resetting signal value so relaxed seems fine.     CP will store-release to set to 0 ; any consumers wait for the 0 value so that is important point to release.